### PR TITLE
Added posts module containing CRUD

### PR DESCRIPTION
**Pull request by uzair-anwar**
In this module, we apply server-side validations for post content and also check the JWT token. we store posts in MySQL database through ORM name sequelize. and also perform other operations like edited, getting all posts,s and deleting a post

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -15,11 +20,17 @@ app.use(express.urlencoded({ extended: true }));
...
     credentials: true,
   })
 );
 
 app.use("/auth", authRoutes);
+app.use("/post", postRoutes);
+
+user.hasMany(post);
+post.belongsTo(user);
+sequelize.sync();
```
Comment (Sufyan-Abbada) [inline server.js]: Use Migrations rather than Schema syncing.

File: src/Controllers/posts.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  try {
+    const posts = await post.findAll();
+    if (posts.length > 0) {
+      res.send({
+        status: 200,
+        result: posts,
+      });
+    } else {
+      res.send({
+        status: 400,
```
Comment (Sufyan-Abbada) [inline src/Controllers/posts.js]: I don't think, 400 is not a better status if posts are not present.

File: src/Controllers/posts.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      });
+    }
+  } catch (error) {
+    res.send({
+      status: 401,
+      message: error.message,
+    });
+  }
+};
+exports.editPost = async (req, res, next) => {
```
Comment (Sufyan-Abbada) [inline src/Controllers/posts.js]: Please be consistent in your code regarding quality.

File: src/Controllers/posts.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  try {
+    const { id } = req.params;
+    const updatedPost = await post.update(
+      { title: req.body.title, content: req.body.content },
```
Comment (Sufyan-Abbada) [inline src/Controllers/posts.js]: Create a logic that, if the user only changes one factor, then the code should not overburden the server with useless replacing.

File: src/Controllers/posts.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      { where: { id: id } }
+    );
+    if (updatedPost[0] === 1) {
```
Comment (Sufyan-Abbada) [inline src/Controllers/posts.js]: Please modify this logic rather than hard code checking of the response

File: src/Controllers/posts.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      status: 400,
+      message: error.message,
+    });
+  }
+};
+
+exports.deletePost = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const rowDeleted = await post.destroy({
```
Comment (Sufyan-Abbada) [inline src/Controllers/posts.js]: Please use self-explanatory variable names.

File: src/Database/migrations/20220809102206-post.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        primaryKey: true,
+      },
+      title: {
+        type: Sequelize.STRING(300),
+        allowNull: false,
+      },
+      content: {
+        type: Sequelize.STRING(1234),
+        allowNull: false,
+      },
```
Comment (Sufyan-Abbada) [inline src/Database/migrations/20220809102206-post.js]: Where is the association of the Posts table with Users?

File: src/Database/models/post.js [hunk 8]
```diff
@@ -1,29 +1,34 @@
 const { Sequelize, DataTypes } = require("sequelize");
```
Comment (Sufyan-Abbada) [inline src/Database/models/post.js]: Please don't use double approaches for a single task within the same project.

**Review by Sufyan-Abbada — CHANGES_REQUESTED** [review]

File: src/Controllers/posts.js (hunk 2 above)
Comment (uzair-anwar) [reply src/Controllers/posts.js]: changed to status code 204

File: src/Database/models/post.js (hunk 8 above)
Comment (uzair-anwar) [reply src/Database/models/post.js]: A user model is necessary to call the function sequelize. That is why we made this. We can make tables with models if we want and also make tables with migrations. If we make tables with migrations this does not mean we have no more needs for models. That why I also made Post or User model

File: src/Controllers/posts.js (hunk 4 above)
Comment (uzair-anwar) [reply src/Controllers/posts.js]: If I make a logic for this, then one extra call for API or one find function of sequelize is called. This means that we run an extra query. This is also an overburden for the server. Updating a field has low cost than a single query that's why I do not create logic for this.

File: src/Database/migrations/20220809102206-post.js (hunk 7 above)
Comment (uzair-anwar) [reply src/Database/migrations/20220809102206-post.js]: Resolved.

File: src/Controllers/posts.js (hunk 5 above)
Comment (uzair-anwar) [reply src/Controllers/posts.js]: Resolved

File: server.js (hunk 1 above)
Comment (uzair-anwar) [reply server.js]: Resolved.

File: server.js [hunk 9]
```diff
@@ -15,11 +20,16 @@ app.use(express.urlencoded({ extended: true }));
 app.use(
   cors({
     origin: ["http://localhost:3000"],
```
Comment (Sufyan-Abbada) [inline server.js]: Localhost should not be present in files. Please move it to `ENV`.

File: server.js [hunk 10]
```diff
@@ -15,11 +20,16 @@ app.use(express.urlencoded({ extended: true }));
...
+    methods: "GET,PUT,PATCH,POST,DELETE",
```
Comment (Sufyan-Abbada) [inline server.js]: Please move these methods in `ENV`.

File: src/Controllers/posts.js (hunk 4 above)
Comment (Sufyan-Abbada) [reply src/Controllers/posts.js]: Good 👍

**Review by Sufyan-Abbada — COMMENTED** [review]
Good work. 
Please don't ever expose sensitive data in your code base. Use `gitignore` and `ENV` for such stuff.

File: server.js (hunk 9 above)
Comment (uzair-anwar) [reply server.js]: Moved

**Review by Sufyan-Abbada — APPROVED** [review]

**Pull request merged**